### PR TITLE
Use alias for nop decomp

### PR DESCRIPTION
**Pull request by SherlockNoMad**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #84727

### CODE REVIEW

**Review by soulitzer — COMMENTED** [review]
Seems OK, test failures seem real though

File: torch/_decomp/decompositions.py [hunk 1]
```diff
@@ -1275,11 +1275,9 @@ def std_decomposition(
 # Questionable decompositions
 # This is only valid if we're running the graph without autograd, such as if the backward pass has been traced.
 # Note that this decomposition causes issues with in-place ops
-@register_decomposition(
-    [aten.detach, aten.lift, aten.lift_fresh, aten.alias], disable_meta=True
-)
+@register_decomposition([aten.detach, aten.lift, aten.lift_fresh], disable_meta=True)
 def nop_decomposition(x):
-    return x
+    return x.alias()
```
[the remark this answers is not in the document]
Comment (SherlockNoMad) [reply torch/_decomp/decompositions.py:1280]: hmm... I am not sure if we want to expose this as public API
I changed it to use aten.alias(x)

File: torch/_decomp/decompositions.py [hunk 2]
```diff
@@ -1275,11 +1275,9 @@ def std_decomposition(
 # Questionable decompositions
 # This is only valid if we're running the graph without autograd, such as if the backward pass has been traced.
 # Note that this decomposition causes issues with in-place ops
-@register_decomposition(
-    [aten.detach, aten.lift, aten.lift_fresh, aten.alias], disable_meta=True
-)
+@register_decomposition([aten.detach, aten.lift, aten.lift_fresh], disable_meta=True)
```
[the remark this answers is not in the document]
Comment (SherlockNoMad) [reply torch/_decomp/decompositions.py:1278]: Unfortunately, this decomposition is still problematic if it's applied before autograd... 

output of aten.detach will set the requires_grad = False, whereas aten.alias will propagate requires_grad's value from input...

[the remark this answers is not in the document]
Comment (soulitzer) [reply torch/_decomp/decompositions.py:1278]: Thanks, that makes sense

File: functorch/test/test_pythonkey.py [hunk 3]
```diff
@@ -590,9 +590,6 @@ def f(x):
         fw_graph, bw_graph = get_fw_bw_graph(f, [torch.randn(5, 5, requires_grad=True)])
         self.assertEqual(get_num_ins_outs(fw_graph), (1, 3))
 
-        ins, outs = get_ins_outs(fw_graph)
-        self.assertEqual(outs[1].target, torch.ops.aten.mm.default)
```
Comment (SherlockNoMad) [inline functorch/test/test_pythonkey.py:594]: With the additional alias node, min_cut_prartitioner is giving a different but still valid result

Comment (Chillee) [reply functorch/test/test_pythonkey.py:594]: Ugh, I think we need to eventually modify these tests to check memory usage instead of exact ops. Oh well.

Comment (Chillee) [reply functorch/test/test_pythonkey.py:594]: But, can we fix this check instead of deleting it? I think you might also need to add it here: https://github.com/pytorch/pytorch/blob/master/functorch/_src/partitioners.py#L278

Or maybe under the inductor section.

**Review by Chillee — APPROVED** [review]
LGTM mostly, modulo the partitioner test
